code/headers/frame_action.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <base_comm.hpp>
+#include <hwlib.hpp>
+
+#include <cstring>
+
+namespace r2d2::robos {
+    template <frame_type Type>
+    class frame_action_c {
+    protected:
+        // set to true if the data from the frame is changed since the last call
+        // of reply_to_data
+        base_comm_c &comm;
+
+        inline static frame_data_t<Type> last_frame = {};
+        inline static bool changed = false;
```

Comment:
Does this need to be static?

Reply:
last_frame needs to be static in order to check for changes for the frame in the constructor
The bool doesn't need to, I will change that

Reply:
partialy fixed in https://github.com/R2D2-2019/RobOS/pull/2/commits/861ab8dbddb7383a995f5ff866feb170625e4275